homeworks/homework_3/arc.cpp

```diff
@@ -0,0 +1,115 @@
+#include <cmath>
+#include <cstdlib>
+#include <ctime>  // for trigger the seed of random number generator
+#include <iostream>
+
+static void genPointsOnUnitSphere(const int N, float *x, float *y, float *z);
+float dotProduct(float *A, float *B, int arr_sz);
+void unitVec(float *A, float *vec, int arr_sz);
+
+int main(int argc, char *argv[]) {
+  // get the size of N
+  int N = std::atoi(argv[1]);
+
+  // sanity check
+  if (argc < 2 || std::isnan(N))
+  {
+    std::cerr << "ERROR: The value of N is either missing or is not a number!" << std::endl;
+    std::exit(1);
+  } else if (N <= 0) {
+    std::cerr << "ERROR: Please choose a value of N that is greater than 0!" << std::endl;
+    std::exit(1);
+  } else {
+
+    // allocate memory for x, y, z
+    float *x = new float [N], *y = new float [N], *z = new float [N];
+
+    genPointsOnUnitSphere(N, x, y, z);
+
+    // determine the extreme arc lengths
+    float length, vec[N][3], n1[3], n2[3];
+
+    // create vectors with x y & z components
+    for (int i = 0; i < N; i++)
+    {
+      vec[i][0] = x[i];
+      vec[i][1] = y[i];
+      vec[i][2] = z[i];
+    }
```

Comment:
You don't have to create these vectors explicitly. You can treat (x[i],y[i], z[i]) as a vector.